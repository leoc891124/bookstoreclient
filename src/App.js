import React from "react";

const App = ()=>{

    return <div>My first app</div>
}

export default App;