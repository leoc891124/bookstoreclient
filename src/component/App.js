import React from "react";
import {BrowserRouter as Router, Route, Routes, Navigate} from "react-router-dom";
import Layout from "./layout/Layout";
import BookContainer from "./book/BookContainer";
import Login from "./user/Login";
import {SnackbarProvider} from "notistack"
//import Auth from "./Auth";
import { useSelector } from "react-redux";
import Register from "./user/Register";
import PrivateRoute from "../PrivateRoute";
//import { useNavigate } from "react-router-dom";
//import {BrowserRouter} from "react-router-dom";

const App = ()=>{
    //const UseNave = useNavigate();
    
    const token = useSelector((state) => state.userredu.token);

    return (
        <SnackbarProvider maxSnack={1} autoHideDuration={1000}>
        <Router>
            <Layout>
                <Routes>
                    <Route path="/login" element={<Login />}></Route>
                    <Route exact path="/" element={<Navigate replace to="/login" />} />
                    <Route path="/dashboard" element={<PrivateRoute> <BookContainer /></PrivateRoute> }></Route>
                    <Route path="/register" element={<Register />}></Route>
                </Routes>
            </Layout>
        </Router>
        </SnackbarProvider>
    );
}

export default App;