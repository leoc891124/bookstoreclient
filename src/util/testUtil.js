import React from 'react';
import { render } from '@testing-library/react';
//import { configureStore } from "@reduxjs/toolkit";
import { Provider } from 'react-redux';
import { setupStore } from "../module/store";
//import bookReducer from "../module/book/bookReducer";
import {SnackbarProvider} from "notistack"


export function renderWithProviders(
  ui,
  {
    preloadedState = { },
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
      ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return { store, ...render(<SnackbarProvider maxSnack={1}>{ui}</SnackbarProvider>, { wrapper: Wrapper, ...renderOptions }) }
}