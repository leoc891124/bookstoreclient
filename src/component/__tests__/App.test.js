/**
 * @jest-environment jsdom
 */
import React from "react";
//import {render} from "@testing-library/react";
import App from "../App";
import {renderWithProviders} from "../../util/testUtil";

describe("App component", ()=>{
    it("should render app with error",()=>{
        const { asFragment} = renderWithProviders(<App />,{});

        expect(asFragment()).toMatchSnapshot();
    })
})