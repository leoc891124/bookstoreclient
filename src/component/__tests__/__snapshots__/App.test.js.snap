// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render app with error 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-0"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1qp5wg9-MuiTypography-root"
          >
            Book Store
          </h6>
        </div>
      </header>
    </div>
    <div
      class="MuiBox-root css-1afphko"
    >
      <form
        autocomplete="off"
        novalidate=""
      >
        <div
          class="makeStyles-wrapper-1 MuiBox-root css-0"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 makeStyles-paper-2 css-1ps6pg7-MuiPaper-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            >
              Book Store Login
            </h4>
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root css-1ezh2tw-MuiFormControl-root-MuiTextField-root"
              data-testid="username-testid"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="username"
                id="username-label"
              >
                Enter username
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="username"
                  name="username"
                  placeholder="Enter username"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Enter username
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root css-1ezh2tw-MuiFormControl-root-MuiTextField-root"
              data-testid="password-testid"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Enter password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  placeholder="Enter password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Enter password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-lll1vm-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Login
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
